src/service/mpris.ts

```diff
@@ -163,10 +202,46 @@ class MprisPlayer extends GObject.Object {
                     this.emit('changed');
                 }
                 catch (e) {
-                    logError(e as Error, `failed to cache ${coverPath}`);
+                    logError(e as Error, `failed to cache ${trackCoverUrl}`);
+                    return "";
                 }
             },
         );
+
+        this._coverCache[hash] = {
+            coverPath,
+            timestamp: Date.now(),
+        };
+
+        if (Object.keys(this._coverCache).length > 100) {
```

Comment:
instead of hardcoding it to 100, there should be an option in App.config for it

Reply:
done